zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java

```diff
@@ -358,6 +361,22 @@ private static void setupClusterManagerServer(ServiceLocator serviceLocator) {
       AuthorizationService authorizationService = serviceLocator.getService(AuthorizationService.class);
       clusterManagerServer.addClusterEventListeners(ClusterManagerServer.CLUSTER_NOTE_EVENT_TOPIC, notebookServer);
       clusterManagerServer.addClusterEventListeners(ClusterManagerServer.CLUSTER_AUTH_EVENT_TOPIC, authorizationService);
+
+      // Since the ClusterInterpreterLauncher is lazy, dynamically generated, So in cluster mode,
+      // when the zeppelin service starts, Create a ClusterInterpreterLauncher object,
+      // This allows the ClusterInterpreterLauncher to listen for cluster events.
+      try {
+        InterpreterSettingManager intpSettingManager = sharedServiceLocator.getService(InterpreterSettingManager.class);
+        RecoveryStorage recoveryStorage = ReflectionUtils.createClazzInstance(
+                conf.getRecoveryStorageClass(),
+                new Class[] {ZeppelinConfiguration.class, InterpreterSettingManager.class},
+                new Object[] {conf, intpSettingManager});
+        recoveryStorage.init();
+        PluginManager.get().loadInterpreterLauncher(InterpreterSetting.CLUSTER_INTERPRETER_LAUNCHER_NAME, recoveryStorage);
```

Comment:
Launcher will be created here https://github.com/apache/zeppelin/blob/master/zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java#L294
Why you do it again here ?

Reply:
Since the `ClusterInterpreterLauncher` is lazy, dynamically generated, So in cluster mode, 
When the user needs to create an interpreter on the cluster (for example: `node A`)
If the `A node` has never created an interpreter and never initialized the `ClusterInterpreterLauncher` object, then the `ClusterInterpreterLauncher` object on the a node will not be able to listen to the create interpreter event sent to it by the cluster.

Therefore, in cluster mode, the `ClusterInterpreterLauncher` object needs to be initialized in advance when the zeppelin server is started.